contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,367 @@
- - For the docker runtime we have a containerd-shim which is a small library that sits in front of a runtime implementation allowing it to be reparented to init, handle reattach from the caller etc. With pod level cgroups containerd-shim can be charged to the pod instead of the machine.
- - If a container exits, all its anonymous pages (tmpfs) gets accounted to the machine (root). With pod level cgroups, that usage can also be attributed to the pod.
- - Let containers share resources - with pod level limits, a pod with a Burstable container and a BestEffort container is classified as Burstable pod. The BestEffort container is able to consume slack resources not used by the Burstable container, and still be capped by the overall pod level limits.
+### CPU allocation
+
+The `BestEffort` cgroup sandbox is statically configured as follows:
+
+```
+ROOT/BestEffort/cpu.shares = 2
+```
+
+This ensures that allocation of CPU time to pods in this QoS class
+is given the lowest priority.
+
+The `Burstable` cgroup sandbox cpu share allocation is dynamic based
+on the set of pods currently scheduled to the node.  
+
+```
+ROOT/Burstable/cpu.shares = max(sum cpu requests of all Burstable pods, 2)
+```
 
-## Design
+The `kubelet` will have a control loop that dynamically adjusts the
+value of the CPU share based on its view of Burstable pods in a
+non-terminal state.  This approach is chosen to reduce overall code
+complexity in the `kubelet`.
 
-High level requirements for the design are as follows:
- - Do not break existing users. Ideally, there should be no changes to the Kubernetes API semantics.
- - Support multiple cgroup managers - systemd, cgroupfs, etc.
+### Memory overcommitment
 
-How we intend to achieve these high level goals is covered in greater detail in the Implementation Plan.
+The `kubelet` highly prioritizes resource utilization, and thus allows
+BestEffort and Burstable pods to potentially consume as many resources
+that are presently available on the node.  In cases where resources become
+scarce, the `kubelet` configures the `oom_killer` and/or pod evictions to
+evict workloads that have the lowest QoS class.
 
-We use the following denotations in the sections below:
+There is latency involved in the pod eviction process which can be a
+cause of concern in latency sensitive servers.  On such servers, we
+want to avoid OOM conditions, and promote the system to steer memory
+consumption on the node to align with the actual amount of
+requested resources for non-terminal Guaranteed and Burstable pods.
 
-For the three QoS classes
-`G⇒ Guaranteed QoS, Bu⇒ Burstable QoS, BE⇒ BestEffort QoS`
+Operators can configure a desired qos memory overcommit (qmo) range using 
+the `--qos-memory-overcommit` flag.  This flag is not enabled
```

Comment:
@vishh -- i see the merit on having a single flag.  will update.  fyi @sjenning we should have a single flag.